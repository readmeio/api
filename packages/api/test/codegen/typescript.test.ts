import chai, { expect } from 'chai';
import { jestSnapshotPlugin } from 'mocha-chai-jest-snapshot';

import Oas from 'oas';
import TSGenerator from '../../src/codegen/typescript';

chai.use(jestSnapshotPlugin());

describe('typescript generator', function () {
  let petstore: Oas;

  beforeEach(async function () {
    petstore = await import('../__fixtures__/simple.oas.json').then(Oas.init);
    await petstore.dereference({ preserveRefAsJSONSchemaTitle: true });
  });

  it('should generate typescript code', async function () {
    const ts = new TSGenerator(petstore);
    const files = await ts.generator();

    expect(files).toMatchSnapshot();
  });
});
