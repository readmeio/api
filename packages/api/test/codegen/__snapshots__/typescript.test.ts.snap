// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typescript generator should generate typescript code 1`] = `
Array [
  Object {
    "index.ts": "import Oas from 'oas';
import APICore from 'api/core';
import definition from '../fixtures/simple.oas.json';

export default class SDK {
  spec: Oas;
  core: APICore;
  authKeys: (number | string)[][] = [];

  constructor() {
    this.spec = Oas.init(definition);
    this.core = new APICore(this.spec, 'api/1.0.0');
  }

  config(config: ConfigOptions) {
    this.core.setConfig(config);
  }

  auth(...values: string[] | number[]) {
    this.core.setAuth(...values);
    return this;
  }

  server(url: string, variables = {}) {
    this.core.setServer(url, variables);
  }

  /**
   * Finds Pets by status
   *
   * Multiple status values can be provided with comma separated strings
   */
  get(path: string, metadata: FindPetsByStatusMetadataParam): Promise<FindPetsByStatus_Response_200>;
  get(path: string, metadata?: Record<string, unknown>): Promise<unknown> {
    return this.core.fetch(path, 'get', metadata);
  }

  /**
   * Finds Pets by status
   *
   * Multiple status values can be provided with comma separated strings
   */
  findPetsByStatus(metadata: FindPetsByStatusMetadataParam): Promise<FindPetsByStatus_Response_200> {
    return this.core.fetch('/pet/findByStatus', 'get', metadata);
  }
}

interface ConfigOptions {
  parseResponse: boolean;
}
export type FindPetsByStatusMetadataParam = {
  /**
   * Status values that need to be considered for filter
   */
  status: (\\"available\\" | \\"pending\\" | \\"sold\\")[];
  [k: string]: unknown;
};
export type FindPetsByStatus_Response_200 = Pet[];
export interface Pet {
  id?: number;
  category?: Category;
  name: string;
  photoUrls: string[];
  tags?: Tag[];
  /**
   * pet status in the store
   */
  status?: \\"available\\" | \\"pending\\" | \\"sold\\";
  [k: string]: unknown;
}
export interface Category {
  id?: number;
  name?: string;
  [k: string]: unknown;
}
export interface Tag {
  id?: number;
  name?: string;
  [k: string]: unknown;
}
",
  },
]
`;
